backend/services/application.py

```diff
@@ -0,0 +1,230 @@
+"""Service that manages applications for the COMP department"""
+
+from typing import List
+from fastapi import Depends
+from sqlalchemy import update, delete
+from sqlalchemy.orm import Session
+from typing import Dict
+from backend.entities import section_application_table
+from backend.entities.application_entity import ApplicationEntity, New_UTA_Entity
+from backend.entities.section_application_table import section_application_table
+from backend.entities.academics.section_entity import SectionEntity
+from backend.models.academics.section import Section
+from backend.models.application_details import (
+    New_UTADetails,
+    UserApplication,
+)
+from backend.models.user import User
+
+from backend.services.exceptions import ResourceNotFoundException
+
+from ..database import db_session
+
+__authors__ = ["Ben Goulet, Abdulaziz Al-Shayef"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class ApplicationService:
+    """ApplicationService is the access layer to TA applications."""
+
+    def __init__(self, session: Session = Depends(db_session)):
+        """Initializes a new ApplicationService.
+
+        Args:
+            session (Session): The database session to use, typically injected by FastAPI.
+        """
+        self._session = session
+
+    def list(self) -> list[New_UTADetails]:
```

Comment:
Adding this comment here because reading through this file, I was a little concerned with return types! Frontend implementation might remedy this (so no worries!), but all of these services are returning `New_UTADetails` - in the case that an application is for a returning member, is that data lost / hidden once this data is transferred to the frontend?